plugins/module_utils/schema.py

```diff
@@ -24,6 +25,15 @@ def __init__(self, mso_module, schema_name, template_name=None, site_name=None):
         if site_name and template_name:
             self.set_site(template_name, site_name)
 
+    @classmethod
+    def with_template_id(cls, mso_module, schema_name, template_name=None, template_id=None):
```

Comment:
why add this as class method? since it has default value of None can't it be added to the constructor?